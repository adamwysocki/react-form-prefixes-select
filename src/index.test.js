// index.test.js
import React from "react";
import PrefixesSelect from "./index";
import renderer from "react-test-renderer";
import Enzyme, { shallow, render, mount } from "enzyme";
import sinon, { stub } from "sinon";
import Adapter from "enzyme-adapter-react-16";

// React 16 Enzyme Adapter
Enzyme.configure({ adapter: new Adapter() });

it("should render", () => {
  const onPrefixSelect = stub();
  const component = renderer.create(<PrefixesSelect onChange={onPrefixSelect} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render without a default option if default option is turned off", () => {
  const onPrefixSelect = stub();
  const wrapper = mount(<PrefixesSelect onChange={onPrefixSelect} hasDefaultOption={false} />);
  const select = wrapper.find("select");
  expect(select.contains(<option value="null">Prefix</option>)).toEqual(false);
});

it("should render with the default option text specified", () => {
  const onPrefixSelect = stub();
  const wrapper = mount(<PrefixesSelect onChange={onPrefixSelect} defaultOptionText={"Prefix ..."} />);
  const select = wrapper.find("select");
  expect(select.contains(<option value="null">Prefix ...</option>)).toEqual(true);
});

it("should call onChange when select option is changed", () => {
  const onChange = sinon.spy();
  const event = { currentTarget: { value: 1 } };
  const wrapper = shallow(<PrefixesSelect onChange={onChange} />);
  wrapper.find("#prefix").simulate("change", event);
  expect(onChange.called).toEqual(true);
});

it("should call not call onChange when default option is selected", () => {
  const onChange = sinon.spy();
  const event = { currentTarget: { value: "null" } };
  const wrapper = shallow(<PrefixesSelect onChange={onChange} />);
  wrapper.find("#prefix").simulate("change", event);
  expect(onChange.called).toEqual(false);
});
