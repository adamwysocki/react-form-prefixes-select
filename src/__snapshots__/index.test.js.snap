// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="sc-bdVaJa dpXQOA"
>
  <select
    className=""
    id="prefix"
    name="prefix"
    onChange={[Function]}
    style={Object {}}
  >
    <option
      value="null"
    >
      Prefix
    </option>
    <option
      value={0}
    >
      1st Lt
    </option>
    <option
      value={1}
    >
      Adm
    </option>
    <option
      value={2}
    >
      Atty
    </option>
    <option
      value={3}
    >
      Brother
    </option>
    <option
      value={4}
    >
      Capt
    </option>
    <option
      value={5}
    >
      Chief
    </option>
    <option
      value={6}
    >
      Cmdr
    </option>
    <option
      value={7}
    >
      Col
    </option>
    <option
      value={8}
    >
      Dean
    </option>
    <option
      value={9}
    >
      Dr
    </option>
    <option
      value={10}
    >
      Elder
    </option>
    <option
      value={11}
    >
      Father
    </option>
    <option
      value={12}
    >
      Gen
    </option>
    <option
      value={13}
    >
      Gov
    </option>
    <option
      value={14}
    >
      Hon
    </option>
    <option
      value={15}
    >
      Lt Col
    </option>
    <option
      value={16}
    >
      Maj
    </option>
    <option
      value={17}
    >
      MSgt
    </option>
    <option
      value={18}
    >
      Mr
    </option>
    <option
      value={19}
    >
      Mrs
    </option>
    <option
      value={20}
    >
      Ms
    </option>
    <option
      value={21}
    >
      Prince
    </option>
    <option
      value={22}
    >
      Prof
    </option>
    <option
      value={23}
    >
      Rabbi
    </option>
    <option
      value={24}
    >
      Rev
    </option>
    <option
      value={25}
    >
      Sister
    </option>
  </select>
</div>
`;
